### DISCUSSION TITLE
about database storing

### DISCUSSION THREAD

**Discussion by Vinaysvinays** [OrchardCMS/OrchardCore#17070] [General]
In Orchard CMS, how is data stored in the database, how are tables formed, and how can data be read and stored from these tables for the Agency theme? If any changes are made to the theme, how is the data stored in the database?

**Comment by hishamco** [comment]
The data handled by [YesSQL](https://github.com/sebastienros/yessql/) all the content stored in the `Document` table and many index tables depend on the enabled features

> how can data be read and stored from these tables

You can use `IContentManager` to access the content, or you could use `ISession` for direct access to the database tables and indices

**Reply by Vinaysvinays** [reply]
In agency theme there is a default inbuilt contact us page if we fill that contact us where will be data stored in the database.
![Screenshot 2024-11-26 164903](https://github.com/user-attachments/assets/59ac08b8-c6b3-4d64-921a-6b1488737a4d)
